app/models/registration.rb

```diff
@@ -176,6 +189,14 @@ def to_be_paid_through_wca?
     !new_record? && (pending? || accepted?) && competition.using_payment_integrations? && outstanding_entry_fees > 0
   end
 
+  def invoice_items_total
+    Money.new(invoice_items.sum(:amount_lowest_denomination), entry_fee.currency)
+  end
+
+  def invoice_items_currency_code
```

Comment:
This is a little awkward but seems like the best I can do until I can store the currency code in the Order? 

Unless I just call `self.competition.currency_code`

Reply:
In line with my other comment above, I think just calling the competition's currency code is fine, since that is what effectively defined every invoice item's currency in your current implementation anyways.

Reply:
Removed - turns out it wasn't used anyway after `invoice_items_total` became a `Money` object